pkg/controller/velerobackup/velerobackup_controller.go

```diff
@@ -169,74 +185,124 @@ func (r *ReconcileBackup) createVeleroBackupObject(namespace string) error {
+			ExcludedResources: defaultExcludedBackupResources,
 		},
 	}
 
 	return r.Create(context.TODO(), backup)
 }
 
-func (r *ReconcileBackup) getModifiedHiveObjectsInNamespace(namespace string) ([]*hiveObject, error) {
-	nsLogger := addNamespaceLoggerFields(r.logger, namespace)
-
-	filteredHiveObjects := []*hiveObject{}
+func addNamespaceLoggerFields(logger log.FieldLogger, namespace string) log.FieldLogger {
+	return logger.WithField("namespace", namespace)
+}
 
-	objList, err := getRuntimeObjects(r, typesToList, namespace)
-	if err != nil {
-		nsLogger.WithError(err).Error("Failed to list hive objects in namespace.")
-		return nil, err
-	}
+func (r *ReconcileBackup) getRuntimeObjects(typesToList []runtime.Object, namespace string) ([]runtime.Object, error) {
+	nsObjects := []runtime.Object{}
 
-	for _, nsObj := range objList {
-		hobj, err := newHiveObject(nsObj, r.logger)
+	for _, t := range typesToList {
+		listObj := t.DeepCopyObject()
+		if err := r.List(context.TODO(), listObj, client.InNamespace(namespace)); err != nil {
+			return nil, err
+		}
+		list, err := meta.ExtractList(listObj)
 		if err != nil {
 			return nil, err
 		}
 
-		if hobj.hasChanged() {
-			filteredHiveObjects = append(filteredHiveObjects, hobj)
+		nsObjects = append(nsObjects, list...)
+	}
+
+	return nsObjects, nil
+}
+
+func (r *ReconcileBackup) getNamespaceCheckpoint(namespace string) (*hivev1.Checkpoint, bool, error) {
+	nsLogger := addNamespaceLoggerFields(r.logger, namespace)
+	cp := &hivev1.Checkpoint{}
+	err := r.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: hiveconstants.CheckpointName}, cp)
+	found := true
+	if err != nil {
+		if errors.IsNotFound(err) {
+			nsLogger.Info("First time backing up this namespace, creating CheckPoint object, ")
+			cp.Name = hiveconstants.CheckpointName
+			cp.Namespace = namespace
+			found = false
+		} else {
+			nsLogger.WithError(err).Error("Failed getting CheckPoint object for namespace.")
+			return nil, false, err
 		}
 	}
 
-	return filteredHiveObjects, nil
+	return cp, found, nil
 }
 
-func (r *ReconcileBackup) updateHiveObjectsLastBackupChecksum(hobjs []*hiveObject) error {
-	for _, hobj := range hobjs {
-		hobjLogger := hobj.addLoggerFields()
+func (r *ReconcileBackup) createOrUpdateNamespaceCheckpoint(cp *hivev1.Checkpoint, found bool) error {
+	nsLogger := addNamespaceLoggerFields(r.logger, cp.Namespace)
+	var err error
 
-		hobj.setChecksumAnnotation()
-		obj := hobj.object
+	if found {
+		err = r.Update(context.TODO(), cp)
+	} else {
+		err = r.Create(context.TODO(), cp)
+	}
 
-		err := r.Update(context.TODO(), obj)
-		if err != nil {
-			hobjLogger.WithError(err).Error("Failed update.")
-			return err // If we error at all, then we should just return. The NS will be requeued in the future.
-		}
+	if err != nil {
+		nsLogger.WithError(err).Error("Failed to create or update CheckPoint object in namespace.")
+		return err
 	}
 
+	// Everything succeeded!
```

Comment:
I'm not sure that this comment is especially helpful. If anything, I would rather see the following.

```
if err != nil {
  nsLogger.WithError(err).Error(...)
}

return err
```